examples/stocks/lib/stock_home.dart

```diff
@@ -161,17 +161,21 @@ class StockHomeState extends State<StockHome> {
           onPressed: _handleMenuShow
         )
       ],
-      tabBar: new TabBar(
-        labels: <TabLabel>[
-          new TabLabel(text: StockStrings.of(context).market()),
-          new TabLabel(text: StockStrings.of(context).portfolio())
-        ]
+      tabBar: new TabBar<StockHomeTab>(
+        tabLabelBuilder: (StockHomeTab tab) {
+          switch(tab) {
+            case StockHomeTab.market:
+              return new TabLabel(text: StockStrings.of(context).market());
+            case StockHomeTab.portfolio:
+              return new TabLabel(text: StockStrings.of(context).portfolio());
+            default:
+              assert(false);
+          }
+        }
       )
```

Comment:
OK, I took my shot at creating a list of objects that represented the tabs and could build a localized string or a TabLabel or a TabView.  This turned out to be even messier than what we've got.